src/docs/quick-start/rust.md

```diff
@@ -0,0 +1,108 @@
+---
+id: rust
+title: 'Rust'
+---
+
+## The Polywrap Client
+
+To use [Wraps](/concepts/wraps) in your app, all you need is the [Polywrap Client](/clients)!
+
+### Rust application boilerplate
+
+We'll be using a simple console application boilerplate for this guide.
+
+Using `cargo init` within a directory, initialize a Rust project.
+
+### Adding Polywrap to your application
+
+Within `cargo.toml`, add dependencies for `polywrap` and `serde`:
+
+```toml title="cargo.toml"
+[dependencies]
+polywrap = "0.1.9-beta.2"
+serde = { version = "1.0.188", features = ["derive"] }
+```
+
+### Invoking your first Wrap
+
+In order to invoke a Wrap, we first need to instantiate the Polywrap Client:
+
+At the top of your `main.rs` file, import `polywrap`:
+
+```rust title="main.rs"
+$snippet: quickstart-rust-use-client
+```
+
+Now, before invoking wraps, we need to define `struct`s that describe the arguments we're passing to the invocation, and, if applicable, the invocation return types.
+
+In our first example, we will be invoking the `sha3_256` function of the SHA3 Wrap, so we will define the following struct:
+
+```rust title="main.rs"
+$snippet: quickstart-rust-invoke-structs
+```
+
+Within our `main` function, we now want to initialize the Polywrap Client:
+```rust title="main.rs"
+$snippet: quickstart-rust-init-client
+```
+
+At this point, you can invoke Wraps! In the example below, we will invoke the SHA3 wrap within our `main` function:
+
+```rust title="main.rs"
+$snippet: quickstart-rust-invoke-client
+```
+
+Running the application using `cargo run`, you should now see the following appear in your console:
+
+```
+ba5a5d5fb7674f5975f0ecd0cd9a2f4bcadc9c04f5ac2ab3a887d8f10355fc38
+```
+
+This is the return value of our invocation.
+
+#### What's going on here?
+
+Using the Polywrap Client, we are invoking the `sha3_256` method of a Wrap found under the [WRAP URI](/concepts/uris) `wrapscan.io/polywrap/sha3@1.0` called the SHA3 Wrap.
```

Comment:
NIT: we can reduce the amount of duplicate text in these docs by using snippets for the contents (and not just the code)

Reply:
I'm actually looking into markdown partials for this kind of thing!

No need to use snippets when Docusaurus' MDX capabilities allow for partial views.